### ISSUE TITLE
Building on Centos 5 fails (O_CLOEXEC not defined) (* with PATCH *)

### ISSUE THREAD

**Issue by kmsquire** [joyent/libuv#400]
Hi,

Building on Centos 5 fails because O_CLOEXEC is not defined (it was introduced in linux kernel 2.6.23, but Centos 5 is still on 2.6.18 -- at least the version I have available).

Here's a possible (simple) patch, although O_CLOEXEC functionality will (obviously) be missing.

``` diff
diff --git a/src/unix/process.c b/src/unix/process.c
index 34740c0..d4b063c 100644
--- a/src/unix/process.c
+++ b/src/unix/process.c
@@ -41,6 +41,10 @@
 extern char **environ;
 #endif

+/* Older systems */
+#ifndef O_CLOEXEC
+# define O_CLOEXEC 0
+#endif

 static void uv__chld(EV_P_ ev_child* watcher, int revents) {
   int status = watcher->rstatus;
```

Kevin

**Comment by bnoordhuis** [comment]
Thanks for pointing that out. Does this patch resolve the issue?

``` diff
diff --git a/src/unix/process.c b/src/unix/process.c
index 10872cc..08e105d 100644
--- a/src/unix/process.c
+++ b/src/unix/process.c
@@ -25,7 +25,6 @@
 #include <assert.h>
 #include <errno.h>
 #include <sys/wait.h>
-#include <fcntl.h> /* O_CLOEXEC, O_NONBLOCK */
 #include <poll.h>
 #include <unistd.h>
 #include <stdio.h>
@@ -108,10 +107,10 @@ int uv__make_pipe(int fds[2], int flags) {
 #if __linux__
   int fl;

-  fl = O_CLOEXEC;
+  fl = UV__O_CLOEXEC;

   if (flags & UV__F_NONBLOCK)
-    fl |= O_NONBLOCK;
+    fl |= UV__O_NONBLOCK;

   if (uv__pipe2(fds, fl) == 0)
     return 0;
```

**Comment by kmsquire** [comment]
Yes, that patch resolves it. Thanks!

Kevin

**Comment by kmsquire** [comment]
I suppose I shouldn't close the issue until you apply the patch... :-)

Kevin

**Comment by bnoordhuis** [comment]
Thanks for testing, Kevin. The fix is committed in 1ebe14e.
